src/safeds/data/tabular/containers/_table.py

```diff
@@ -2193,26 +2198,64 @@ def plot_histograms(self) -> Image:
         >>> image = table.plot_histograms()
         """
         import matplotlib.pyplot as plt
+        import numpy as np
         import pandas as pd
-        import seaborn as sns
 
-        col_wrap = min(self.number_of_columns, 3)
+        n_cols = min(3, len(self.column_names))
+        n_rows = (len(self.column_names) + n_cols - 1) // n_cols
 
-        data = pd.melt(self._data.map(lambda value: str(value)), value_vars=self.column_names)
-        grid = sns.FacetGrid(data=data, col="variable", col_wrap=col_wrap, sharex=False, sharey=False)
-        grid.map(sns.histplot, "value")
-        grid.set_xlabels("")
-        grid.set_ylabels("")
-        grid.set_titles("{col_name}")
-        for axes in grid.axes.flat:
-            axes.set_xticks(axes.get_xticks())
-            axes.set_xticklabels(axes.get_xticklabels(), rotation=45, horizontalalignment="right")
-        grid.tight_layout()
-        fig = grid.fig
+        if n_cols == 1 and n_rows == 1:
+            one_col = True
+            fig, axs = plt.subplots(1, 1, tight_layout=True, figsize=(3, 3))
+        else:
+            one_col = False
+            fig, axs = plt.subplots(n_rows, n_cols, tight_layout=True, figsize=(n_cols * 3, n_rows * 3))
+
+        col_names = self.column_names
+        for col, ax in zip(col_names, axs.flatten() if not one_col else [axs]):
```

Comment:
```suggestion
        for col_name, ax in zip(col_names, axs.flatten() if not one_col else [axs]):
```

Clarifies that this is just the name of the column, not the column itself. You could also loop over the columns, though, since you get the column anyway in the next line. The name of the column can be retrieved using `column.name`.